server/src/main/java/org/opensearch/search/aggregations/metrics/MaxAggregator.java

```diff
@@ -104,6 +104,21 @@ public ScoreMode scoreMode() {
         return valuesSource != null && valuesSource.needsScores() ? ScoreMode.COMPLETE : ScoreMode.COMPLETE_NO_SCORES;
     }
 
+    @Override
+    protected boolean tryPrecomputeAggregationForLeaf(LeafReaderContext ctx) throws IOException {
+        CompositeIndexFieldInfo supportedStarTree = getSupportedStarTree(this.context.getQueryShardContext());
+        if (supportedStarTree != null) {
+            if (parent != null && subAggregators.length == 0) {
+                // If this a child aggregator, then the parent will trigger star-tree pre-computation.
+                // Returning NO_OP_COLLECTOR explicitly because the getLeafCollector() are invoked starting from innermost aggregators
+                return true;
```

Comment:
So a problem I noticed in my star-tree changes was I cannot throw `CollectionTerminatedException` because of the hierarchy of how aggregators get called when in a nested structure.

Basically if a nested metric aggregator throws a `CollectionTerminatedException` the parent's `getLeafCollector()` call also terminates - so we might not do any pre-computation in parent aggregator.

Should be easily verifiable if that is the case (that it is mishandled) - the date histogram star-tree tests will end up failing for sure in that case.

Reply:
I verified this by tests and manually running through queries as well against your branch. I guess this scenario is handled gracefully.

Reply:
Yeah... I was wondering about that, but it looks like the tests are passing. In particular, if I run

```
./gradlew ':server:test' --tests "org.opensearch.search.aggregations.startree.**"
```

They all pass, including the date histogram tests. 🤷